base/multidimensional.jl

```diff
@@ -863,3 +863,193 @@ If `dim` is specified, returns unique regions of the array `itr` along `dim`.
         @nref $N A d->d == dim ? sort!(uniquerows) : (1:size(A, d))
     end
 end
+
+"""
+    C, ia, ib, ic = uniqueind(itr, dim)
+
+A function that operates similiarly to `unique(itr,dim)` but returns multiple
+output arguments having the following properties:
+
+C - the unique elements of the array `itr` along the selected dimension `dim`
+ia - a Vector{Int} of indices such that:
+     `itr[ia] == C` returns `true` if `itr` is a one-dimensional array and `dim == 1`
+     `itr[ia,:] == C` returns `true` if `itr` is a two-dimensional array and `dim == 1`
+     `itr[:,ia] == C` returns `true` if `itr` is a two-dimensional array and `dim == 2`
+     `itr[:,:,ia] == C` returns `true if `itr` is a three-dimensional array and `dim == 3`
+     and so forth for higher dimensional arrays.
+ib - a Vector{Vector{Int}} where each Vector{Int} contains the indices associated with the
+     individual entries of `C` along the dimension `dim`
+ic - A Vector{Int} of indices such that:
+     `C[ic] == itr` returns `true` if `itr` is a one-dimensional array and `dim == 1`
+     `C[ic,:] == itr` returns `true` if `itr` is a two-dimensional array and `dim == 1`
+     `C[:,ic] == itr` returns `true` if `itr` is a two-dimensional array and `dim == 2`
+     `C[:,:,ic] == itr` returns `true if `itr` is a three-dimensional array and `dim == 3`
+     and so forth for higher dimensional arrays.
+
+Examples:
+
+julia> itr = [1;2;3;2;3;5;6;5;7;1];
+julia> C, ia, ib, ic = uniqueind(itr,1)
+([1,2,3,5,6,7],[1,2,3,6,7,9],[[1,10],[2,4],[3,5],[6,8],[7],[9]],[1,2,3,2,3,4,5,4,6,1])
+julia> C[ic] == itr
+true
+julia> itr[ia] == C
+true
+julia> ib
+6-element Array{Array{Int64,1},1}:
+ [1,10]
+ [2,4]
+ [3,5]
+ [6,8]
+ [7]
+ [9]
+
+julia> D = rand(Int,2,3,1);
```

Comment:
Better always use the same values for doctests I think. That will allow running them as tests, and arrays with no duplicated value would be less interesting to understand how this works.